src/network/clientpackethandler.cpp

```diff
@@ -1209,8 +1209,14 @@ void Client::handleCommand_HudSetParam(NetworkPacket* pkt)
 
 	if (param == HUD_PARAM_HOTBAR_ITEMCOUNT && value.size() == 4) {
 		s32 hotbar_itemcount = readS32((u8*) value.c_str());
-		if (hotbar_itemcount > 0 && hotbar_itemcount <= HUD_HOTBAR_ITEMCOUNT_MAX)
+		if (hotbar_itemcount > 0 && hotbar_itemcount <= HUD_HOTBAR_ITEMCOUNT_MAX) {
+			u32 main_size = player->inventory.getList("main")->getSize();
+			if ((u32)hotbar_itemcount > main_size)
+				hotbar_itemcount = main_size;
+			if (player->getWieldIndex() >= hotbar_itemcount)
+				player->setWieldIndex(hotbar_itemcount - 1);
```

Comment:
Unnecessary. This action is likely overwritten by `Game::processItemSelection` which specifies `runData.new_playeritem` to update the index when necessary (i.e. on resize). Hence range checks might be more useful there.